drivers/hmis/app/jobs/hmis/auto_exit_job.rb

```diff
@@ -0,0 +1,80 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Hmis
+  class AutoExitJob < BaseJob
+    def perform
+      Hmis::Hud::Project.all.each do |project|
+        config = Hmis::AutoExitConfig.config_for_project(project)
+        next unless config.present?
+
+        project.enrollments.each do |enrollment|
+          most_recent_contact = if HudUtility2024.residential_project_type_ids.include?(project.project_type)
+            enrollment.services.where(record_type: 200).order(:date_provided).last
+          else
+            [
+              enrollment.services.order(:date_provided).last,
+              enrollment.custom_services.order(:date_provided).last,
+              enrollment.current_living_situations.order(:information_date).last,
+              enrollment.custom_assessments.order(:assessment_date).last,
+            ].compact.max_by { |entity| contact_date_for_entity(entity) }
+          end
+
+          next unless most_recent_contact.present?
+
+          most_recent_contact_date = contact_date_for_entity(most_recent_contact)
+          next unless (Date.today - most_recent_contact_date).to_i >= config.length_of_absence_days
+
+          auto_exit(enrollment, most_recent_contact)
+        end
+      end
+    end
+
+    def configs
+      @configs ||= Hmis::AutoExitConfig.all
+    end
+
+    private
+
+    def auto_exit(enrollment, most_recent_contact)
+      exit_date = contact_date_for_entity(most_recent_contact)
+      exit_date += 1.day if most_recent_contact.is_a?(Hmis::Hud::Service) && most_recent_contact.record_type == 200
+      Hmis::Hud::Exit.create!(
+        personal_id: enrollment.personal_id,
+        enrollment_id: enrollment.enrollment_id,
+        data_source_id: enrollment.data_source_id,
+        user_id: Hmis::Hud::User.system_user(data_source_id: enrollment.data_source_id).user_id,
+        exit_date: exit_date,
+        destination: 30,
+        auto_exited: DateTime.now,
+      )
+      assessment = Hmis::Hud::CustomAssessment.new_with_defaults(
+        enrollment: enrollment,
+        user: Hmis::Hud::User.system_user(data_source_id: enrollment.data_source_id),
+        form_definition: Hmis::Form::Definition.find_definition_for_role(:EXIT, project: enrollment.project),
```

Comment:
This isn't necessary, we can leave it empty. No form definition was used to perform the assessment, so there is no need to store the reference to the one that was used. When we migrate in assessments, we also leave it blank. Maybe just don't use the `new_with_defaults` function
```suggestion
```